src/datamodel/game/board.rs

```diff
@@ -0,0 +1,5 @@
+use crate::datamodel::game::quest_objective::QuestObjective;
+
+pub struct Board {
+    pub quest_objectives: Vec<QuestObjective>,
+}
```

Comment:
Since Board is an static object on game time that doesn't change it composition, but only the state of its components. Shouldn't this struct have a constructor that builds the board with the sequence of objectives?
Shouldn't it be the case for an `iterator impl`?

https://doc.rust-lang.org/book/ch13-02-iterators.html#the-iterator-trait-and-the-next-method

Reply:
Right now we decided not to add any method to these classes since we actually don't know how we're going to use them. The only place in the whole code base where I see instantiation of any of the classes we wrote so far is the `GameCreator` class. At this point it is not 100% clear to me how the instantiation would benefit from extra constructors in these classes, so I prefer not to write those until we don't need them, to make sure we actually do what makes more sense for the implementation we need, and not what's already available from a default implementation.

Reply:
ok, let's agree that development is not mature enough to add complex stuff and move along ASAP. So, I'll resolve.
Nevertheless, I will create an issue for this add-on anyway #11 .